x-pack/plugins/observability/public/hooks/use_get_filtered_rule_types.ts

```diff
@@ -6,10 +6,13 @@
  */
 
 import { useMemo } from 'react';
+import { ES_QUERY_ID } from '@kbn/rule-data-utils';
 import { usePluginContext } from './use_plugin_context';
 
 export function useGetFilteredRuleTypes() {
   const { observabilityRuleTypeRegistry } = usePluginContext();
 
-  return useMemo(() => observabilityRuleTypeRegistry.list(), [observabilityRuleTypeRegistry]);
+  return useMemo(() => {
+    return [...observabilityRuleTypeRegistry.list(), ES_QUERY_ID];
```

Comment:
Why does the Elasticseaarch query rule have a separate group? I think it should be under the observability group; I am not sure if it was introduced in this PR, though 🤔 

cc @katrin-freihofner 

![image](https://github.com/elastic/kibana/assets/12370520/35dd822e-3dca-4455-96b5-3fe2f2b398e6)

Reply:
I think we can create a ticket to fix that later on but I understood that it will be like that at first

Reply:
I agree, it should not be in a separate section. We have a ticket related to that: https://github.com/elastic/actionable-observability/issues/83